ts/features/barcode/hooks/useCameraPermissionStatus.ts

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useEffect, useState } from "react";
+import { AppState, Linking } from "react-native";
+import { Camera, CameraPermissionStatus } from "react-native-vision-camera";
+import { isAndroid } from "../../../utils/platform";
+
+/**
+ * Hook to handle camera permission status with platform specific behavior
+ */
+export const useCameraPermissionStatus = () => {
+  const [cameraPermissionStatus, setCameraPermissionStatus] =
+    useState<CameraPermissionStatus>();
+
+  /**
+   * Opens the system prompt to ask camera permission
+   */
+  const requestCameraPermission = useCallback(async () => {
+    const permissions = await Camera.requestCameraPermission();
+    setCameraPermissionStatus(permissions);
+  }, []);
+
+  /**
+   * Opens the system settings to allow user to change the camera permission
+   */
+  const openCameraSettings = useCallback(() => {
+    void Linking.openSettings();
+  }, []);
+
+  /**
+   * Checks the camera permission on mount
+   *
+   * **Note:** On android devices the app starts with a "denied" permission status,
+   * which does not necessarily mean that the permission was denied by the user.
+   * We need to request the permission to check if the user has denied it.
+   */
+  useEffect(() => {
+    const permission = Camera.getCameraPermissionStatus();
+    if (isAndroid && permission === "denied") {
+      // Waits 500ms to ensure navigation is completed
+      setTimeout(requestCameraPermission, 500);
```

Comment:
Isn't there a more robust approach to handle this?

Reply:
Yes, there is a more robust approach but it's a bit more complicated. 
The permission request blocks the JS thread so we need to wait for the navigation animation to end in order to correctly request the permissions.
I have found a solution in this commit: https://github.com/pagopa/io-app/pull/6687/commits/384ab44f437c0c70ebfb6d12969006cc5afc841b
Using the `useIsFocused` hook in addition of the `transitionEnd` event, we can request the camera permission when the user is in the correct screen. 
Let me know!

Reply:
Super! I did it a test and it works great.